lib/src/client/transport/xhr_transport.dart

```diff
@@ -115,17 +108,15 @@ class XhrTransportStream implements GrpcTransportStream {
     if (!_validateResponseState()) {
       return;
     }
-    _incomingMessages.add(GrpcMetadata(_request.responseHeaders));
```

Comment:
It looks like package:http exposes an extension method on XmlHttpRequest to make `responseHeaders` available:

https://github.com/dart-lang/http/blob/b97b8dc22ea808c4fbd63f73abd7af8ecf694323/pkgs/http/lib/src/browser_client.dart#L124

Using it would avoid needing to implement HTTP header parsing below in _parseHeaders.

Reply:
This extension method seems to be implicity and private, so if a name is given to this extension i may use `import 'package:http/src/browser_client.dart'` and `_request.responseHeaders`.

https://stackoverflow.com/questions/59092361/import-extension-method-from-another-file-in-dart

Reply:
I filed https://github.com/dart-lang/web/pull/298